external-providers/yq-external-provider/pkg/yq_provider/service_client.go

```diff
@@ -0,0 +1,317 @@
+package yq_provider
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/konveyor/analyzer-lsp/jsonrpc2"
+	"github.com/konveyor/analyzer-lsp/lsp/protocol"
+	"github.com/konveyor/analyzer-lsp/provider"
+	"go.lsp.dev/uri"
+	"golang.org/x/mod/semver"
+	"gopkg.in/yaml.v2"
+)
+
+type yqServiceClient struct {
+	rpc        *jsonrpc2.Conn
+	cancelFunc context.CancelFunc
+	log        logr.Logger
+	cmd        *exec.Cmd
+
+	config       provider.InitConfig
+	capabilities protocol.ServerCapabilities
+}
+
+type Release struct {
+	Name string `json:"name"`
+}
+
+const APIVERSION = "apiVersion"
+const KIND = "kind"
+const GITHUB_K8S_API_URL = "https://api.github.com/repos/kubernetes/kubernetes/releases"
+
+var default_k8s_version = "v1.28.4"
```

Comment:
we may want to think of a way to make this more dynamic

Reply:
Should I add a config struct and then assign those values form a json file or yaml file?